components/server/src/workspace/workspace-starter.ts

```diff
@@ -838,11 +838,18 @@ export class WorkspaceStarter {
             await this.tryEnableConnectionLimiting(featureFlags, user, billingTier);
             await this.tryEnablePSI(featureFlags, user, billingTier);
 
-            let usageAttributionId = await this.userService.getWorkspaceUsageAttributionId(user, workspace.projectId);
             // if the workspace has been created in an organization, we need to use the organization's attribution ID
-            if (workspace.organizationId) {
-                const org = await this.teamDB.findTeamById(workspace.organizationId);
-                usageAttributionId = AttributionId.create(org!);
+            let usageAttributionId: AttributionId | undefined = workspace.organizationId
+                ? {
+                      kind: "team",
+                      teamId: workspace.organizationId,
+                  }
+                : undefined;
+            if (!usageAttributionId && !user.additionalData?.isMigratedToTeamOnlyAttribution) {
+                usageAttributionId = await this.userService.getWorkspaceUsageAttributionId(user);
+            }
+            if (usageAttributionId === undefined) {
+                throw new Error("No usage attribution ID found");
             }
```

Comment:
It would help readability to wrap this in a function which either always returns an attribution ID, or throws. The block here would become a bit more readable and the extra branching is kept outside of the main flow.

Reply:
I've improved the readability a bit by introducing a factory method for creating attr ids by organizationId. Would like to keep the rest of the logic inlined and remove it entirely soon.